youtube_dl/extractor/qqmusic.py

```diff
@@ -77,14 +104,24 @@ def _real_extract(self, url):
 
         formats = []
         for format_id, details in self._FORMATS.items():
-            formats.append({
-                'url': 'http://cc.stream.qqmusic.qq.com/%s%s.%s?vkey=%s&guid=%s&fromtag=0'
-                       % (details['prefix'], mid, details['ext'], vkey, guid),
-                'format': format_id,
-                'format_id': format_id,
-                'preference': details['preference'],
-                'abr': details.get('abr'),
-            })
+            video_url = 'http://cc.stream.qqmusic.qq.com/%s%s.%s?vkey=%s&guid=%s&fromtag=0' \
+                        % (details['prefix'], mid, details['ext'], vkey, guid)
+            req = HEADRequest(video_url)
```

Comment:
Could you use [`_check_formats` or `_is_valid_url`](https://github.com/rg3/youtube-dl/blob/621ed9f5f4d9d82659272ebe01e740e9196fad61/youtube_dl/extractor/common.py#L771-L792)?